javatests/arcs/showcase/typevariables/VariableTest.kt

```diff
@@ -0,0 +1,64 @@
+package arcs.showcase.typevariables
+
+import arcs.core.allocator.Allocator
+import arcs.core.host.*
+import arcs.core.storage.api.DriverAndKeyConfigurator
+import arcs.core.storage.driver.RamDisk
+import arcs.core.storage.driver.RamDiskDriverProvider
+import arcs.jvm.host.ExplicitHostRegistry
+import arcs.jvm.host.JvmHost
+import arcs.jvm.host.JvmSchedulerProvider
+import arcs.jvm.util.testutil.FakeTime
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import kotlin.coroutines.EmptyCoroutineContext
+
+@RunWith(JUnit4::class)
+@ExperimentalCoroutinesApi
+class VariableTest {
+
+    class TestJvmProdHost(
+        schedulerProvider: SchedulerProvider,
+        vararg particles: ParticleRegistration
+    ): JvmHost(schedulerProvider, *particles), ProdHost
+
+    @Test
+    fun shop_redactsSku() = runBlocking {
+        RamDisk.clear()
+        DriverAndKeyConfigurator.configureKeyParsers()
+        RamDiskDriverProvider()
+
+        val hostRegistry = ExplicitHostRegistry()
+        val schedulerProvider = JvmSchedulerProvider(EmptyCoroutineContext)
+
+        hostRegistry.registerHost(
+            TestJvmProdHost(schedulerProvider,
+                ::OrderIngestion.toRegistration(),
+                ::SkuRedactor.toRegistration(),
+                ::Consumer.toRegistration()
+            )
+        )
+
+        val allocator = Allocator.create(
```

Comment:
You should be able to replace all this setup with ShowcaseEnvironment JUnitRule now.

Reply:
Cool! I'll look into it